src/Groups/types.jl

```diff
@@ -275,8 +357,19 @@ end
 # Construct an Oscar group wrapping the GAP group `obj`
 # *and* compatible with a given Oscar group `G`.
 
+sub_type(T::Type) = T
+sub_type(::Type{PcGroup}) = SubPcGroup
+sub_type(G::GAPGroup) = sub_type(typeof(G))
+
+# `_oscar_group(obj, G)` is used to create the subgroup of `G`
+# that is described by the GAP group `obj`;
 # default: ignore `G`
-_oscar_group(obj::GapObj, G::T) where T <: GAPGroup = T(obj)
+function _oscar_group(obj::GapObj, G::GAPGroup)
+  S = sub_type(G)(obj)
+  @assert GAP.Globals.FamilyObj(S.X) === GAP.Globals.FamilyObj(G.X)
+  return S
+end
+#T better rename to _oscar_subgroup?
```

Comment:
I agree: `_oscar_subgroup` would be better

Reply:
O.k., I will rename the two-argument version.